### PR TITLE
Orb slam3 mac

### PR DESCRIPTION
**Pull request by match08**
Mac support for ORB-SLAM3

### CODE REVIEW

File: include/LoopClosing.h [hunk 1]
```diff
@@ -47,8 +47,12 @@ class LoopClosing
 public:
 
     typedef pair<set<KeyFrame*>,int> ConsistentGroup;    
+    // typedef map<KeyFrame*,g2o::Sim3,std::less<KeyFrame*>,
+    //     Eigen::aligned_allocator<std::pair<const KeyFrame*, g2o::Sim3> > > KeyFrameAndPose;
+    
+    //Mac support for ORB-SLAM3
     typedef map<KeyFrame*,g2o::Sim3,std::less<KeyFrame*>,
-        Eigen::aligned_allocator<std::pair<const KeyFrame*, g2o::Sim3> > > KeyFrameAndPose;
+            Eigen::aligned_allocator<std::pair<KeyFrame *const, g2o::Sim3> > > KeyFrameAndPose;
```
Comment (soeroesg) [inline include/LoopClosing.h:55]: @match08 const KeyFrame* is not the same as KeyFrame *const and we need the first one here!

Comment (runer112) [reply include/LoopClosing.h:55]: The change looks correct to me. I encountered the same issue building on Windows, and this change was the fix for me as, well.

**Comment by match08** [comment]
> Mac support for ORB-SLAM3

Mac support for ORB-SLAM3 Tested: [Viewer and AR Example to MacOS](https://github.com/match08/ORB_SLAM3/commit/404da9ef5f8829a12b627ec164c8bea8f50ffbb5)

**Comment by IamShubhamGupto** [comment]
@jdtardos 
could this code live on a `macOS` branch? I was able to replicate orb slam3 on my m1 Mac Pro

**Pull request closed without merging**